descope/auth/enchantedlink.go

```diff
@@ -0,0 +1,100 @@
+package auth
+
+import (
+	"net/http"
+
+	"github.com/descope/go-sdk/descope/errors"
+)
+
+type enchantedLink struct {
+	authenticationsBase
+}
+
+func (auth *enchantedLink) SignIn(identifier, URI string, r *http.Request, loginOptions *LoginOptions) (*EnchantedLinkResponse, error) {
+	var pswd string
+	var err error
+	if identifier == "" {
+		return nil, errors.NewInvalidArgumentError("identifier")
+	}
+	if loginOptions.IsJWTRequired() {
+		pswd, err = getValidRefreshToken(r)
+		if err != nil {
+			return nil, errors.InvalidStepupJwtError
+		}
+	}
+	httpResponse, err := auth.client.DoPostRequest(composeEnchantedLinkSignInURL(), newMagicLinkAuthenticationRequestBody(identifier, URI, true, loginOptions), nil, pswd)
+	if err != nil {
+		return nil, err
+	}
+	return getPendingRefFromResponse(httpResponse)
+}
+
+func (auth *enchantedLink) SignUp(identifier, URI string, user *User) (*EnchantedLinkResponse, error) {
+	if identifier == "" {
+		return nil, errors.NewInvalidArgumentError("identifier")
+	}
+	if user == nil {
+		user = &User{}
+	}
+	if len(user.Email) == 0 {
+		user.Email = identifier
+	}
+
+	httpResponse, err := auth.client.DoPostRequest(composeEnchantedLinkSignUpURL(), newMagicLinkAuthenticationSignUpRequestBody(MethodEmail, identifier, URI, user, true), nil, "")
+	if err != nil {
+		return nil, err
+	}
+	return getPendingRefFromResponse(httpResponse)
+}
+
+func (auth *enchantedLink) SignUpOrIn(identifier, URI string) (*EnchantedLinkResponse, error) {
+	if identifier == "" {
+		return nil, errors.NewInvalidArgumentError("identifier")
+	}
+	httpResponse, err := auth.client.DoPostRequest(composeEnchantedLinkSignUpOrInURL(), newMagicLinkAuthenticationRequestBody(identifier, URI, true, nil), nil, "")
+	if err != nil {
+		return nil, err
+	}
+	return getPendingRefFromResponse(httpResponse)
+}
+
+func (auth *enchantedLink) GetSession(pendingRef string, w http.ResponseWriter) (*AuthenticationInfo, error) {
+	var err error
+	httpResponse, err := auth.client.DoPostRequest(composeGetSession(), newAuthenticationGetMagicLinkSessionBody(pendingRef), nil, "")
+	if err != nil {
+		if err == errors.UnauthorizedError {
+			return nil, errors.MagicLinkUnauthorized
```

Comment:
Shouldn't it be errors.EnchantedLinkUnauthorized?

Reply:
fixed